### PR TITLE
Add storageclass validator and validate unique default

### PR DESCRIPTION
**Pull request by chrisho**
**Problem:**
Multiple StorageClass can be set as "default"

**Solution:**
Add a StorageClass validator to the harvester/webhook, if it is checked that a default sc already exists, it cannot be configured, need to reset the existing one.

**Related Issue:**
https://github.com/harvester/harvester/issues/2863

**Test plan:**
https://github.com/harvester/harvester/issues/2863

### CODE REVIEW

File: pkg/webhook/resources/storageclass/validator.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	return validateSetUniqueDefault(newSC, scList)
+}
+
+func validateSetUniqueDefault(newSC *storagev1.StorageClass, scList []*storagev1.StorageClass) error {
+	// if new storage class haven't storageclass.kubernetes.io/is-default-class annotation or the value is false.
+	if v, ok := newSC.Annotations[util.AnnotationIsDefaultStorageClassName]; !ok || v == "false" {
+		return nil
+	}
+
```
Comment (futuretea) [inline pkg/webhook/resources/storageclass/validator.go:55]: prefer to move v.storageClassCache.List logic to here (line 64)

and merge two validateSetUniqueDefault function

File: pkg/webhook/resources/storageclass/validator.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	for _, sc := range scList {
+		if sc.Name == newSC.Name {
+			continue
+		}
+
+		// return set default error
+		// when find another have storageclass.kubernetes.io/is-default-class annotation and value is true.
+		if v, ok := sc.Annotations[util.AnnotationIsDefaultStorageClassName]; ok && v == "true" {
+			return werror.NewInvalidError("reset existing default storage class first", "")
```
Comment (guangbochen) [inline pkg/webhook/resources/storageclass/validator.go]:
```suggestion
			return werror.NewInvalidError("default storage class %s already exists, please reset it first", sc.Name)
```

**Review by guangbochen — APPROVED** [review]
LGTM, and please squash the commits.

**Review by futuretea — APPROVED** [review]
LGTM

**Pull request merged**